plugin/src/automation/lift/lift.ml

```diff
@@ -414,30 +420,28 @@ let lift_app_lazy_delta c env f args lift_rec sigma =
   let l = get_lifting c in
   let sigma, f' = lift_rec env sigma c f in
   let sigma, args' = map_rec_args lift_rec env sigma c args in
-  if (not (equal f f')) || l.is_fwd || Array.length args = 0 || is_opaque c f then
+  (* TODO really need to get below working more generallly; why does it infinite recurse with is_fwd for some kinds of ornaments? what is it not considering? *)
```

Comment:
Address this. It looks like perhaps infinite recursion was due to unpreprocessed match statements. But we should still investigate and see if we can relax this at all.